detect/detect.go

```diff
@@ -36,8 +36,17 @@ const (
 var (
 	newLineRegexp = regexp.MustCompile("\n")
 	isWindows     = runtime.GOOS == "windows"
+	tmpDir        string
 )
 
+func init() {
+	var err error
+	tmpDir, err = os.MkdirTemp("", "gitleaks-*")
```

Comment:
@rgmz lemme know what you think of this approach. I considered creating a "tmpdirmanager" struct but that seemed like overkill.